File: Backend/Domain/Entities/Product.cs
public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Sku { get; set; } = default!;     
    public void AssignSku(string sku)
{
    if (!string.IsNullOrWhiteSpace(Sku))
        throw new InvalidOperationException("SKU already assigned");
    Sku = sku;
}
    public string Description { get; set; }
    public Guid CategoryId { get; set; }
    public Category Category { get; set; }
    public bool IsActive { get; set; }=true;
}

File: FrontEnd/Types/product.ts
export interface ProductDto {
  id: string;
  name: string;
  description: string;
  category:string;
  sku:string
}
export interface CreateProductDto {
  name: string;
  description: string;
  price: number;
}

export interface UpdateProductDto {
  name?: string;
  description?: string;
  price?: number;
}